engines/wintermute/base/gfx/opengl/base_surface_opengl3d.cpp

```diff
@@ -167,36 +231,35 @@ bool BaseSurfaceOpenGL3D::comparePixel(int x, int y, byte r, byte g, byte b, int
 }
 
 bool BaseSurfaceOpenGL3D::startPixelOp() {
-	glBindTexture(GL_TEXTURE_2D, tex->getTextureName());
+	_imageData = new uint8[4 * _texWidth * _texHeight]();
```

Comment:
Potential leak-opportunity, if startPixelOp is called more than once, without calling endPixelOp inbetween. Perhaps we should add a check that _imageData is nullptr here, and error out in that case?

Reply:
In case _imageData is not the nullptr (which means startPixelOp was already called before), startPixelOp just returns success now. That should make it leak safe (together with the deletion inside the destructor).

Reply:
It would make it leak-safe, but potentially also hide what I assume would be an error, perhaps it would be better to at least warn in that case, or perhaps assert?

Reply:
Sorry, I misunderstood your first comment. I put in an assert now, much more bulletproof I guess.